lib/inspec/rspec_json_formatter.rb

```diff
@@ -237,23 +260,42 @@ def current_control_infos
   end
 
   def current_control_summary(fails, skips)
-    sum_info = [
-      (fails.length > 0) ? "#{fails.length} failed" : nil,
-      (skips.length > 0) ? "#{skips.length} skipped" : nil,
-    ].compact
-
     summary = @current_control[:title]
-    unless summary.nil?
-      return summary + ' (' + sum_info.join(' ') + ')' unless sum_info.empty?
-      return summary
-    end
 
-    return sum_info.join(' ') if @current_control[:results].length != 1
-
-    fails.clear
-    skips.clear
-    c = @current_control[:results][0]
-    c[:code_desc].to_s + c[:message].to_s
+    res = @current_control[:results]
+    if res.length == 1
+      # Single test - be nice and just print the exception message if the test
+      # failed. No need to say "1 failed".
+      fails.clear
+      skips.clear
+      c = res[0]
+      # If it's an anonymous control, just go with the only description
+      # available for the underlying test.
+      summary = c[:code_desc].to_s unless summary
+      summary + c[:message].to_s
+    elsif res.length == 0
+      # Empty control block - if it's anonymous, there's nothing we can do.
+      # Is this case even possible?
+      summary = "Empty anonymous control" unless summary
```

Comment:
I think you want something like: `summary || "Empty anonymous control"` otherwise this will return `nil` in the case that summary is already defined.